webpack/assets/javascripts/react_app/redux/reducers/hosts/powerStatus/powerStatus.test.js

```diff
@@ -11,30 +11,39 @@ import {
 } from './powerStatus.fixtures';
 
 import reducer from './index';
+import { testReducerEqualityWithFixture } from '../../../../common/testHelpers';
 
 describe('powerStatus reducer', () => {
-  it('should return the initial state', () => {
-    expect(reducer(undefined, {})).toEqual(initialState);
-  });
-
-  it('should handle HOST_POWER_STATUS_REQUEST', () => {
-    expect(reducer(initialState, {
-      type: types.HOST_POWER_STATUS_REQUEST,
-      payload: request,
-    })).toEqual(stateBeforeResponse);
-  });
-
-  it('should handle HOST_POWER_STATUS_SUCCESS', () => {
-    expect(reducer(stateBeforeResponse, {
-      type: types.HOST_POWER_STATUS_SUCCESS,
-      payload: response,
-    })).toEqual(stateAfterSuccess);
-  });
-
-  it('should handle HOST_POWER_STATUS_FAILURE', () => {
-    expect(reducer(stateBeforeResponse, {
-      type: types.HOST_POWER_STATUS_FAILURE,
-      payload: { error, id: request.id },
-    })).toEqual(stateAfterFailure);
-  });
+  const fixtures = {
+    'should return the initial state': {
+      prev: undefined,
+      action: {},
+      next: initialState,
+    },
+    'should handle HOST_POWER_STATUS_REQUEST': {
+      prev: initialState,
+      action: {
+        type: types.HOST_POWER_STATUS_REQUEST,
+        payload: request,
+      },
+      next: stateBeforeResponse,
+    },
+    'should handle HOST_POWER_STATUS_SUCCESS': {
+      prev: stateBeforeResponse,
+      action: {
+        type: types.HOST_POWER_STATUS_SUCCESS,
+        payload: response,
+      },
+      next: stateAfterSuccess,
+    },
+    'should handle HOST_POWER_STATUS_FAILURE': {
+      prev: stateBeforeResponse,
+      action: {
+        type: types.HOST_POWER_STATUS_FAILURE,
+        payload: { error, item: { id: request.id } },
```

Comment:
could we store the entire resource under item ?

Reply:
By saying "entire resource" are you talking about the `request`?
Basically you want to change this line to
```
payload: { error, item: request }
```
Right?

Reply:
After a second look, only the id is stored via `ajaxRequestAction`